lib/command.js

```diff
@@ -2193,4 +2231,14 @@ function getCommandAndParents(startCommand) {
   return result;
 }
 
+/**
+ * @param {*} value
+ * @returns {boolean}
+ * @api private
+ */
+
+function isThenable(value) {
```

Comment:
I did this in a my recent PR too, so apologies I am having second thoughts about where to put the code. Buckle up for some subtle considerations.

An unexported global function is not available to authors who subclass Command, and in general I don't want to explicitly export internal utility functions. I have wondered about static methods, but they can't be overridden (not that I want to explicitly support overriding it as such!).

So what I have done in a couple of places is make some common code a private method as the best compromise (at least when used by only a single class).

With the optional chaining it is pretty compact so could just be inlined, but I do like naming the test.

Reply:
I don't think we want to let users overwrite the thenable check, do we? If not, I think the function is placed right where it needs to be.

Reply:
There are a few other global functions which have not caused problems to date, and easy workaround (copy). I'll put the paranoia aside!

Reply:
I don't even know why you think global functions in a module should cause any problem, they are not available in user code even if the entire module is `import`ed (and not just some variables) because [strict mode is always enabled in modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode?retiredLocale=de#strict_mode_for_modules).